### PR TITLE
Ppd42x

### PR DESCRIPTION
**Pull request by pi4homez**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#<esphome-docs PR number goes here>
**Pull request in [esphome-core](https://github.com/esphome/esphome-core) with C++ framework changes (if applicable):** esphome/esphome-core#<esphome-core PR number goes here>

## Checklist:
  - [x ] The code change is tested and works locally.
  - [x ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [not yet ] Documentation added/updated in [esphomedocs](https://github.com/OttoWinter/esphomedocs).

### CODE REVIEW

File: esphome/components/sensor/ppd42x.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from esphome.cpp_helpers import gpio_input_pin_expression, gpio_output_pin_expression, \
+    setup_component
+from esphome.cpp_types import App
+
+
+Ppd42xSensorComponent = sensor.sensor_ns.class_('Ppd42xSensorComponent',
+                                                sensor.PollingSensorComponent)
+
+
+PLATFORM_SCHEMA = cv.nameable(sensor.SENSOR_PLATFORM_SCHEMA.extend({
```
Comment (OttoWinter) [inline esphome/components/sensor/ppd42x.py]:
```suggestion
PLATFORM_SCHEMA = sensor.PLATFORM_SCHEMA.extend({
```

Same as in -core, this class shouldn't publish both pm10 and pm2.5 in the same sensor.

File: esphome/components/sensor/ppd42x.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    cv.GenerateID(): cv.declare_variable_id(Ppd42xSensorComponent),
+    vol.Optional(CONF_PM_10_0): vol.All(pins.internal_gpio_input_pin_schema,
+                                        pins.validate_has_interrupt),
```
Comment (OttoWinter) [inline esphome/components/sensor/ppd42x.py]: - Remove `pins.validate_has_interrupt)`
- Rename the option here to `pm_10_0_pin`.

File: esphome/components/sensor/ppd42x.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    vol.Optional(CONF_PM_2_5):  vol.All(pins.internal_gpio_input_pin_schema,
+                                        pins.validate_has_interrupt),
+    vol.Required(CONF_TIMEOUT): cv.positive_time_period_microseconds,
```
Comment (OttoWinter) [inline esphome/components/sensor/ppd42x.py]: This shouldn't be required.

File: tests/test1.yaml [hunk 4]
```diff
@@ -5,8 +5,8 @@ esphome:
   # Use this for testing while developing:
   # Note the travis check for esphome PRs will fail until the
   # esphome-core PR has been merged.
-  # esphome_core_version:
-  #   local: ~/path/to/esphome-core
+  esphome_core_version:
+    local: /home/pi4homez/github2/esphome-core
```
Comment (OttoWinter) [inline tests/test1.yaml]: Remove these changes after having tested the changes

**Review by OttoWinter — APPROVED** [review]

File: tests/test1.yaml (hunk 4 above)
Comment (pi4homez) [reply tests/test1.yaml]: Hi
I'm new on git and do not know how to test my implemetantion :(
Do i need to do it localy or may do it on hass dev esphome?
thanks for advise

File: esphome/components/sensor/ppd42x.py (hunk 2 above)
Comment (pi4homez) [reply esphome/components/sensor/ppd42x.py]: not clear Rename the option here to pm_10_0_pin., what do i have to do?
Thanks

Comment (OttoWinter) [reply esphome/components/sensor/ppd42x.py]: Create a new constant called `CONF_PM_10_0_PIN = 'pm_10_0_pin'` above `PLATFORM_SCHEMA`, same for pm_2_5_pin

File: tests/test1.yaml [hunk 5]
```diff
@@ -502,6 +502,15 @@ sensor:
     update_interval: 5min
     rx_only: false
 
+  - platform: ppd42x
+    timeout: 30000 # mandatory
+    pm_2_5:
+      name: "SDS011 PM2.5"
+    pm_10_0:
+      name: "SDS011 PM10.0"
+    update_interval: 5min
```
Comment (OttoWinter) [inline tests/test1.yaml]: I'm pretty sure this test will fail validation run as-is. timeout needs a time unit and pm_2_5 accepts a pin, not name as-is.

**Comment by pi4homez** [comment]
Help @otto , my request is alwas in error in the step continuous-integration/travis-ci/pr — The Travis CI build failed.
can you advise please ?
thanks

**Pull request closed without merging**